### ISSUE TITLE
Using referrer is not working

### ISSUE THREAD

**Issue by lneninger** [johngodley/redirection#544]
To help track problems please follow this template. If you can provide screenshots then that will help.

*Please replace this template with your own details!*

### Summary
The referrer is not working at all.

Explain what you are trying to do. If you want to suggest a new feature then explain what you want it to do
I trying to set a condition based on the referrer. Basically if the request came from an specific referrer the redirection not occurs but if the referrer match the user is redirected to the specificated url.
I tried using referrer as plain text and as regular expression without success.

### Actual Behavior
The redirection rule is only matching the url pattern. The referrer always fail

Explain what actually happened. If suggesting a new feature then explain what currently happens.

### Steps to reproduce
Set this configuration
**Source URL:**  /(.*)/\?radar as Regex
**Url and Referrer**
**Referrer:** facebookexternalhit as regex or
facebookexternalhit/1.1 (+http://www.facebook.com/externalhit_uatext.php) as plain text
**When matched:** Redirect to URL with HTTP Code 301
**Matched Target:** empty
**Unmatched Target:** https://www.microsoft.com 

 Go to facebook and link to http://<dominio>/?radar

_The url should be http://<dominio>/?radar because the url and referrer matchs(see the plugin log)_

Explain the steps involved to reproduce the problem
Just install the plugin and configure

### Environment
Production

Please list your browser, browser version, version of WordPress and Redirection plugin.
Chrome

**Comment by johngodley** [comment]
Sorry for taking such a long time to reply.

I created the redirect as described above. I added it as:

/(.*)/\?radar

This is because `?` is a regular expression character and needs escaping.

I then:

`curl -v -I http://latest.local/thing/\?radar`

The response is as expected:

`Location: https://www.microsoft.com`

I then try:

`curl -v -I http://latest.local/thing/\?radar -H "Referer: facebookexternalhit"`

The response is also as expected.
